Failed to build

this is the configure
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for pcap_open_live in -lpcap... yes
checking for pcap_set_immediate_mode in -lpcap... yes
checking for cap_get_proc in -lcap... no
configure: creating ./config.status

Make failed
gcc -O3 -march=native -DSN_SELECTION_RTT  -I ./include        src/edge.c  libn2n.a  -o edge
libn2n.a(random_numbers.o): In function `n2n_seed':
random_numbers.c:(.text+0x214): undefined reference to `_rdseed64_step'
collect2: error: ld returned 1 exit status
make: *** [edge] Error 1



Please，What should I do？

This will not work, it will show the following error：
src/aes.c: In function ‘aes_internal_key_setup’:
src/aes.c:300:5: error: ‘for’ loop initial declarations are only allowed in C99 mode
     for(int i = 1; i < ctx->Nr; ++i) {
     ^
src/aes.c:300:5: note: use option -std=c99 or -std=gnu99 to compile your code
make: *** [src/aes.o] Error 1

If I use -std=c99, it also does not work：

src/transform_aes.c: In function ‘n2n_transop_aes_init’:
src/transform_aes.c:226:5: error: unknown type name ‘u_char’
     const u_char *encrypt_key = (const u_char *)conf->encrypt_key;
     ^
src/transform_aes.c:226:5: error: unknown type name ‘u_char’
src/transform_aes.c:245:5: warning: passing argument 2 of ‘setup_aes_key’ from incompatible pointer type [enabled by default]
     return setup_aes_key(priv, encrypt_key, encrypt_key_len);
     ^
src/transform_aes.c:178:12: note: expected ‘const uint8_t *’ but argument is of type ‘const int *’
 static int setup_aes_key (transop_aes_t *priv, const uint8_t *password, ssize_t password_len) {

--- a/src/random_numbers.c
+++ b/src/random_numbers.c
@@ -142,6 +142,7 @@ uint64_t n2n_seed (void) {
 
     // __RDSEED__ ist set only if architecturual feature is set, e.g. compile with -march=native
 #ifdef __RDSEED__
+#if __GNUC__ > 4
     for(i = 0; i < RND_RETRIES; i++) {
         if(_rdseed64_step((unsigned long long*)&seed)) {
             // success!
@@ -153,6 +154,7 @@ uint64_t n2n_seed (void) {
     if(i == RND_RETRIES) {
         traceEvent(TRACE_ERROR, "n2n_seed was not able to get a hardware generated random number from RDSEED.");
     }
+#endif
 #endif
 
 #ifdef WIN32

